spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MapTable.scala

```diff
@@ -124,10 +126,30 @@ class MapTable[T: ClassTag](
   }
 }
 
-object MapTable {
+object MapTable extends ContainerSerializable {
   def apply[@specialized(Float, Double) T: ClassTag](
     module: AbstractModule[_ <: Activity, _ <: Activity, T] = null
   )(implicit ev: TensorNumeric[T]) : MapTable[T] = {
     new MapTable[T](module)
   }
+
+  override def loadModule[T: ClassTag](model : BigDLModule)
+                                      (implicit ev: TensorNumeric[T]) : ModuleData[T] = {
+    val moduleData = super.loadModule(model)
+    val mapTable = moduleData.module.asInstanceOf[MapTable[T]]
+    require(mapTable.modules.size >=1, "sub module should not be empty")
+    mapTable.add(mapTable.modules(0))
+    moduleData
+  }
+
+  override def serializeModule[T: ClassTag](module : ModuleData[T])
+                                           (implicit ev: TensorNumeric[T]) : BigDLModule = {
+    val mapTable = module.module.asInstanceOf[MapTable[T]]
+    val subModules = mapTable.modules
+    require(subModules.size >=1, "sub module should not be empty")
+    val singleModule = subModules(0)
+    mapTable.modules.clear()
+    mapTable.modules.append(singleModule)
```

Comment:
If you check the implementation you could see only one module exist there and copies 'n' times while forwarding , n is the input size. Actually I could directly use container serializer  to store all sub modules, this implementation is just to save storage cost.